src/Expander.hs

```diff
@@ -382,18 +382,18 @@ initializeKernel = do
   where
     typePrims :: [(Text, (SplitTypePtr -> Syntax -> Expand (), TypePatternPtr -> Syntax -> Expand ()))]
     typePrims =
-      [ ("Syntax", Prims.baseType TSyntax)
-      , ("String", Prims.baseType TString)
-      , ("Signal", Prims.baseType TSignal)
+      [ ("Syntax", Prims.baseType $ TyF TSyntax [])
+      , ("String", Prims.baseType $ TyF TString [])
+      , ("Signal", Prims.baseType $ TyF TSignal [])
       , ("->", Prims.arrowType)
       , ("Macro", Prims.macroType)
-      , ("Type", Prims.baseType TType)
+      , ("Type", Prims.baseType $ TyF TType [])
       ]
 
     funPrims :: [(Text, Scheme Ty, Value)]
     funPrims =
       [ ( "string=?"
-        , Scheme 0 $ Ty (TFun (Ty TString) (Ty (TFun (Ty TString) (Prims.primitiveDatatype "Bool" []))))
+        , Scheme 0 (tString `tFun` tString `tFun` Prims.primitiveDatatype "Bool" [])
```

Comment:
note: infix `tFun` is right-associative, like `(->)`.

Reply:
Ak! Scary! Fixity declarations for backticked names is a fairly obscure Haskell feature, isn't it?

I'd probably have made it be `[Ty] -> Ty -> Ty` instead, to remove all doubt at call sites, but this does work.

Reply:
E.g. `tFun [tString, tString] (tPrimData "Bool" [])`

Reply:
Sure, if you prefer. That doesn't work with `TyF`, so I've changed the method from `tFun :: arg -> arg -> a` to `tFun1 :: arg -> arg -> a`, and I have defined `tFun :: [Ty] -> Ty -> Ty` in terms of it.